specs/StatusBarMessage.md

```diff
@@ -0,0 +1,106 @@
+<!-- USAGE
+  * Fill in each of the sections (like Background) below
+  * Wrap code with `single line of code` or ```code block```
+  * Before submitting, delete all <!-- TEMPLATE marked comments in this file,
+    and the following quote banner:
+-->
+# Background
+The browser has a Status bar that displays text when hovering over a link, or performing some activity. Currently, 
+developers are able to opt in to disable showing the status bar through the browser 
+settings.
+
+Developers would also like to be able to opt in to intercept the messages which would 
+normally be displayed by the Status bar, and show it using thier own custom UI. 
+# Description
+We propose a new event for WebView2 that would allow developers to 
+listen for Status bar updates which are triggered by activity on the WebView, and then handle those updates however they want in their applications.
```

Comment:
They're _status_ updates, not _status bar_ updates, right? Whether the host of the control wants to display the message in a status bar is it's own choice. So the "StatusBar" name in the APIs should just be "Status".

Reply:
Yes that sounds good.
* `StatusMessage` for the property
* `StatusMessageChanged` for the property changed event

Reply:
I liked `StatusBarMessage` because this is "the message that is displayed in the status bar (or would be displayed if we had one). If you are rendering a replacement status bar, this is the message you should be showing in it."

`StatusMessage` is not clear about what types of status messages you get.

Reply:
Also, should it be `StatusBarText` rather than `-Message`?

Reply:
That's a good point. Looking at precedent:
* `window.status` (Deprecated. This was for setting the status bar text)
* `window.statusbar.visible` (Deprecated. The statusbar object had only a visible property)
* `WebBrowserControl.StatusBarText`
* `WebBrowser.StatusText`

Using 'status bar' instead of 'status' I think does make it clearer. Saying its a status **bar** isn't prescribing the form of UI so much as saying what the kind of status is, since this is a thing in browsers.

Let's do:
* `StatusBarText` 
* `StatusBarTextChanged`

Reply:
The control never has a status bar though. There are control features and host features, and status bar is up to the host. Also, doesn't this text align with `window.status`?

Reply:
For precedent it seems not super helpful since in the web world they were confused with both status and statusbar, and then similar in our mshtml controls we also had both StatusBar and Status. They do agree that we should use Text instead of Message though.

We have another API CoreWebView2Settings.IsStatusBarEnabled which defaults to true and uses the Edge browser's status bar UI which is a floating box in the bottom left. 

![image](https://user-images.githubusercontent.com/2541336/126200438-84fa61b0-54f3-4766-aab5-bdb3d1445385.png)

Reply:
So status _bar_ is already a control concept?

Reply:
(I believe chromium calls their UI control (in the screenshot above) a status bubble. But the more general term used across browsers is status bar.) And yes that status bar is on by default in WebView2. It can be turned off via IsStatusBarEnabled and then additionally we're adding the ability for the end dev to make their own UI control with this API.